src/prefect/automations.py

```diff
@@ -0,0 +1,114 @@
+from typing import Optional
+from uuid import UUID
+
+from pydantic import Field
+from typing_extensions import Self
+
+from prefect.client.utilities import get_or_create_client
+from prefect.events.schemas.automations import AutomationCore
+from prefect.utilities.asyncutils import sync_compatible
+
+
+class Automation(AutomationCore):
+    id: Optional[UUID] = Field(default=None, description="The ID of this automation")
+
+    @classmethod
+    @sync_compatible
+    async def create(cls: Self, automation: Self) -> Self:
+        """
+        Create a new automation.
+
+        auto_to_create = Automation(
+        name="woodchonk",
+        trigger=EventTrigger(
+            expect={"animal.walked"},
+            match={
+                "genus": "Marmota",
+                "species": "monax",
+            },
+            posture="Reactive",
+            threshold=3,
+            within=timedelta(seconds=10),
+        ),
+        actions=[CancelFlowRun()]
+        )
+        created_automation = Automation.create(auto_to_create)
+        """
+        client, _ = get_or_create_client()
+        automation = AutomationCore(**automation.dict(exclude={"id"}))
+        automation_id = await client.create_automation(automation=automation)
+        automation = await client.read_automation(automation_id=automation_id)
+        automation = Automation(**automation.dict())
+        return automation if automation else None
+
+    @sync_compatible
+    async def update(self: Self) -> Self:
+        """
+        Updates an existing automation.
+        auto = Automation.read(id = 123)
+        auto.name = "new name"
+        auto.update()
+        """
+
+        client, _ = get_or_create_client()
+        automation = AutomationCore(**self.dict(exclude={"id", "owner_resource"}))
+        automation = await client.update_automation(
+            automation_id=self.id, automation=automation
+        )
```

Comment:
if its not meant to return anything we can remove that return type

but its strange that we assign to `automation` if we're not using that, so maybe
```suggestion
    @sync_compatible
    async def update(self: Self):
        """
        Updates an existing automation.
        auto = Automation.read(id = 123)
        auto.name = "new name"
        auto.update()
        """

        client, _ = get_or_create_client()
        automation = AutomationCore(**self.dict(exclude={"id", "owner_resource"}))
        await client.update_automation(
            automation_id=self.id, automation=automation
        )
```